### PR TITLE
Test notification foreground and tap detection in objc app

### PR DESCRIPTION
**Pull request by guperrot**
Test app changes to verify https://github.com/MicrosoftDocs/appcenter-docs-pr/pull/875.
Will do swift after this one is approved to avoid having to address same feedbacks twice.

### CODE REVIEW

File: Sasquatch/SasquatchObjC/AppDelegate.m [hunk 1]
```diff
@@ -383,7 +389,33 @@ - (BOOL)distribute:(MSDistribute *)distribute
...
+  completionHandler(UNNotificationPresentationOptionNone);
+}
+
+// iOS 10 and later, asks the delegate to process the user's response to a
+// delivered notification.
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center
+    didReceiveNotificationResponse:(UNNotificationResponse *)response
+             withCompletionHandler:(void (^)(void))completionHandler
+    API_AVAILABLE(ios(10.0)) {
+  self.notificationResponse = [response actionIdentifier];
```
Comment (MatkovIvan) [inline Sasquatch/SasquatchObjC/AppDelegate.m]: Since [UNNotificationDismissActionIdentifier](https://developer.apple.com/documentation/usernotifications/unnotificationdismissactionidentifier)
> is delivered only if the notification’s category object was configured with the customDismissAction option

and there are only 2 system actions + we are not supporting categories with custom actions, it looks redundant.
Lets just set flag to `NO` here:
```objc
self.didReceiveNotificationInForeground = NO;
```

File: Sasquatch/SasquatchObjC/AppDelegate.m [hunk 2]
```diff
@@ -383,7 +389,33 @@ - (BOOL)distribute:(MSDistribute *)distribute
...
+// When this callback is called, this disables the other callback that MSPush
+// handles.
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center
+       willPresentNotification:(UNNotification *)notification
+         withCompletionHandler:
+             (void (^)(UNNotificationPresentationOptions options))
+                 completionHandler API_AVAILABLE(ios(10.0)) {
+  self.didReceiveNotificationInForeground = YES;
+  [MSPush didReceiveRemoteNotification:notification.request.content.userInfo];
+  completionHandler(UNNotificationPresentationOptionNone);
```
Comment (MatkovIvan) [inline Sasquatch/SasquatchObjC/AppDelegate.m]: Let's expose it to UI to test system alert in the foreground case.

Comment (clpolet) [reply Sasquatch/SasquatchObjC/AppDelegate.m]: What do you want to expose ?

File: Sasquatch/SasquatchObjC/AppDelegate.m (hunk 1 above)
Comment (clpolet) [reply Sasquatch/SasquatchObjC/AppDelegate.m]: Do you mean adding a new flag `self.userDidTapNotification` instead of the `notificationResponse` ?

File: Sasquatch/SasquatchObjC/AppDelegate.m (hunk 2 above)
Comment (MatkovIvan) [reply Sasquatch/SasquatchObjC/AppDelegate.m]: There is "Alert" option. In this case the callback is called twice - here and when user clicked

Comment (MatkovIvan) [reply Sasquatch/SasquatchObjC/AppDelegate.m]: So, either picker or just switch "show system notification on foreground push"

File: Sasquatch/SasquatchObjC/AppDelegate.m (hunk 1 above)
Comment (MatkovIvan) [reply Sasquatch/SasquatchObjC/AppDelegate.m]: Just setting `didReceiveNotificationInForeground` to `NO` would be enough

File: Sasquatch/SasquatchObjC/AppDelegate.m (hunk 2 above)
Comment (guperrot) [reply Sasquatch/SasquatchObjC/AppDelegate.m]: I don't understand this feedback, can you rephrase?

File: Sasquatch/SasquatchObjC/AppDelegate.m (hunk 1 above)
Comment (guperrot) [reply Sasquatch/SasquatchObjC/AppDelegate.m]: If background the notification is always tapped.
If foreground the notification is never tapped.
We could indeed simplify this + the docs.
Let's keep the boolean for both app and docs.

File: Sasquatch/SasquatchObjC/AppDelegate.m [hunk 3]
```diff
@@ -26,22 +27,18 @@
...
     MSAnalyticsDelegate,
 #endif
-    MSCrashesDelegate, MSDistributeDelegate, MSPushDelegate>
+    MSCrashesDelegate, MSDistributeDelegate, MSPushDelegate,
+    UNUserNotificationCenterDelegate>
 
 @property(nonatomic) MSAnalyticsResult *analyticsResult;
 
+@property(nonatomic) BOOL didTapNotification;
 @end
```
Comment (jaelim-ms) [inline Sasquatch/SasquatchObjC/AppDelegate.m]: Empty line before `@end`
